jina/peapods/zmq/__init__.py

```diff
@@ -255,14 +255,23 @@ def send_message(self, msg: 'Message'):
         self.msg_sent += 1
 
         if o_sock == self.out_sock and self.in_sock_type == zmq.DEALER:
-            self._send_idle()
+            self._send_idle_to_router()
 
-    def _send_idle(self):
-        """Tell the upstream router this dealer is idle """
-        msg = ControlMessage('IDLE', pod_name=self.name, identity=self.identity)
+    def _send_control_to_router(self, command):
+        msg = ControlMessage(command, pod_name=self.name, identity=self.identity)
         self.bytes_sent += send_message(self.in_sock, msg, **self.send_recv_kwargs)
         self.msg_sent += 1
-        self.logger.debug(f'idle and i {self.identity} told the router')
+        self.logger.debug(
+            f'control message {command} with id {self.identity} is sent to the router'
+        )
+
+    def _send_idle_to_router(self):
+        """Tell the upstream router this dealer is idle """
+        self._send_control_to_router('IDLE')
```

Comment:
Watch out, there is a risk of 'recursion'. SENT_IDLE_ROUTER calls SEND_MESSAGE which calls SEND_IDLE_ROUTER.

I think this should be refactored

Reply:
ah no, my bad